### PR TITLE
MGMT-14159: Use imageDigestSources in install-config instead of ICSP

### PR DESCRIPTION
**Pull request by bfournie**
In https://issues.redhat.com/browse/OCPNODE-1258 imageContentSources has been deprecated and replaced by imageDigestSources. This change sets imageDigestSources in the install-config.yaml that is generated for the installer. As a result the ImageDigestMirrorSet resource is created in the installed cluster and the ImageContentSourcePolicy is not.

```
$ oc describe ImageDigestMirrorSet
Name:         image-digest-mirror
Namespace:
Labels:       <none>
Annotations:  <none>
API Version:  config.openshift.io/v1
Kind:         ImageDigestMirrorSet
Metadata:
  Creation Timestamp:  2023-12-11T14:58:25Z
  Generation:          1
  Resource Version:    1512
  UID:                 2d851151-e931-4c05-9fa2-d8b223bcad5a
Spec:
  Image Digest Mirrors:
    Mirrors:
      virthost.ostest.test.metalkube.org:5000/localimages/local-release-image
    Source:  quay.io/openshift-release-dev/ocp-v4.0-art-dev
    Mirrors:
      virthost.ostest.test.metalkube.org:5000/localimages/local-release-image
    Source:  registry.ci.openshift.org/ocp/release
Status:
Events:  <none>

$ oc get ImageContentSourcePolicy
No resources found
```

## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/installcfg/installcfg.go [hunk 1]
```diff
@@ -178,6 +178,14 @@ type ImageContentSource struct {
 	Source  string   `json:"source"`
 }
 
+type ImageDigestSource struct {
```
Comment (danielerez) [inline internal/installcfg/installcfg.go:181]: Can we remove the ImageContentSource struct now? (line 176)

Comment (bfournie) [reply internal/installcfg/installcfg.go:181]: So it seems that for now we should still keep it in installConfig struct and marked as Deprecated to match the installer https://github.com/openshift/installer/blob/master/pkg/types/installconfig.go#L148, similar to how other deprecated fields are handled. The field is planned for removal in 4.17.

Comment (danielerez) [reply internal/installcfg/installcfg.go:181]: Oh ok, make sense. So can we just add a comment on the struct that it's deprecated?

Comment (bfournie) [reply internal/installcfg/installcfg.go:181]: Sure. Added a comment.

**Pull request merged**